### DISCUSSION TITLE
Any way to auto-migrate from Portainer?

### DISCUSSION THREAD

**Discussion by parasiteoflife** [moghtech/komodo#161] [Q&A]
Is the only way to stop all stacks, add them in komodo and then starting them again?
Would be great to have some way to auto-import from portainer maintaining the create and update dates of the stacks/containers

**Comment by mbecker20** [comment]
You don't need to stop the Stacks. In fact the best way is to deploy Komodo alongside Portainer, and move over incrementally. 

You will need to make a new Stack in Komodo matching the name of the docker compose project (find with `docker compose ls`), and give Komodo a way to access the compose file contents (in UI defined, Files on Server mode, or Git Repo based files).

To do this in bulk, you will likely find it easier to write out all the stack configuration into a Resource Toml File, and sync all the definitions into Komodo. See https://komo.do/docs/sync-resources#stack. If you have used Terraform, the idea is similar.

Unfortunately there is no tool to migrate directly from Portainer. This would be an excellent community contribution.

**Reply by mpkossen** [reply]
Can you define multiple stacks in a single Resource Toml file? That would definitely speed up the deployment of a large amount of stacks through a resource sync.

**Reply by mbecker20** [reply]
Yes, you can define as many resources (including Stacks) as you want in a single file.
